libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -437,6 +444,16 @@ void HAL_Linux::setup_signal_handlers() const
     sigaction(SIGINT, &sa, NULL);
 }
 
+void HAL_Linux::set_cpu_affinity(const char *str) const 
+{
+    cpu_set_t cpu_affinity;
+    if (!utilInstance.parse_cpu_set(str, &cpu_affinity)) {
+        fprintf(stderr, "Invalid option for cpu-afffinity: %s\n", str);
```

Comment:
error message about wrong option is clearer if together where you are parsing them.

Reply:
Done
pi@raspberrypi:~ $ sudo /home/pi/arducopter -A  udp:192.168.1.144:14550:bcast -B /dev/serial0 -c 
Raspberry Pi 4 Model B Rev 1.2. (intern: 4)
Unknown option '?'
pi@raspberrypi:~ $ sudo /home/pi/arducopter -A  udp:192.168.1.144:14550:bcast -B /dev/serial0 -c 1,
Raspberry Pi 4 Model B Rev 1.2. (intern: 4)
Invalid option for cpu-affinity:  missing cpu number
pi@raspberrypi:~ $ sudo /home/pi/arducopter -A  udp:192.168.1.144:14550:bcast -B /dev/serial0 -c 1:
Raspberry Pi 4 Model B Rev 1.2. (intern: 4)
Invalid option for cpu-affinity:  did you means separator - e.g. 1-3
pi@raspberrypi:~ $ sudo /home/pi/arducopter -A  udp:192.168.1.144:14550:bcast -B /dev/serial0 -c 1-
Raspberry Pi 4 Model B Rev 1.2. (intern: 4)
Invalid option for cpu-affinity:  missing end cpu number
pi@raspberrypi:~ $ sudo /home/pi/arducopter -A  udp:192.168.1.144:14550:bcast -B /dev/serial0 -c 1-2
Raspberry Pi 4 Model B Rev 1.2. (intern: 4)
Loaded defaults from /home/pi/ardupilot.parm
Loaded defaults from /home/pi/ardupilot.parm